enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java

```diff
@@ -20,49 +20,69 @@
 package org.neo4j.causalclustering.logging;
 
 import java.io.PrintWriter;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.util.Date;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
 
+import org.neo4j.causalclustering.core.consensus.RaftMessages;
 import org.neo4j.kernel.lifecycle.LifecycleAdapter;
 
 import static java.lang.String.format;
 import static java.lang.String.valueOf;
 
 public class BetterMessageLogger<MEMBER> extends LifecycleAdapter implements MessageLogger<MEMBER>
 {
+    private enum Direction {
+        INFO( "---" ),
+        OUTBOUND( "-->" ),
+        INBOUND( "<--" );
+
+        public final String arrow;
+
+        Direction( String arrow )
+        {
+            this.arrow = arrow;
+        }
+    }
+
     private final PrintWriter printWriter;
-    private DateFormat dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern( "yyyy-MM-dd HH:mm:ss.SSS" );
```

Comment:
This does not include timezone information, correct? This information is often useful so we can correlate between the messages in this log and debug/neo4j logs.

Reply:
Fixed

Reply:
Updated to include time zone provided by clock. We don't seem to have a common clock in the Modules, I used `Clocks.systemClock`

Reply:
we have PlatformModule::clock thingy

Reply:
Not in 3.1. It's also not used for logging, which afaics is created via a builder specifying UTC timezone.